playground/environment-react-ssr/vite.config.ts

```diff
@@ -0,0 +1,104 @@
+import { fileURLToPath } from 'node:url'
+import {
+  type Connect,
+  type Plugin,
+  type PluginOption,
+  createServerModuleRunner,
+  defineConfig,
+} from 'vite'
+
+export default defineConfig((env) => ({
+  clearScreen: false,
+  appType: 'custom',
+  plugins: [
+    vitePluginSsrMiddleware({
+      entry: '/src/entry-server',
+      preview: fileURLToPath(
+        new URL('./dist/server/index.js', import.meta.url),
+      ),
+    }),
+    {
+      name: 'global-server',
+      configureServer(server) {
+        Object.assign(globalThis, { __globalServer: server })
+      },
+    },
+  ],
+  // [feedback] no automatic process.env.NODE_ENV replacement applied for build?
+  define:
+    env.command === 'build'
+      ? {
+          'process.env.NODE_ENV': `"production"`,
+        }
+      : {},
+  environments: {
+    client: {
+      build: {
+        minify: false,
+        sourcemap: true,
+        outDir: 'dist/client',
+      },
+    },
+    ssr: {
+      build: {
+        outDir: 'dist/server',
+        // [feedback]
+        // is this still meant to be used?
+        // for example, `ssr: true` seems to make `minify: false` automatically
+        // and also externalization.
+        ssr: true,
+        rollupOptions: {
+          input: {
+            index: '/src/entry-server',
+          },
+        },
+      },
+    },
+  },
+
+  // [feedback] should preview automatically pick up environments.client.build.outDir?
```

Comment:
We didn't work yet on the preview side, but I was thinking we should have a `PreviewEnvironment` and make preview work for the whole app in the same way that we are now letting `vite build` build the whole app. This is something that was requested by framework authors before and we have been slowly making Preview work closer to the dev server. I think vite should work at the app level in all commands.